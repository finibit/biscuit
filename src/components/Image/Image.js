import React from 'react'
import styled from 'styled-components'
import styles from '../../styles'
import types from '../../types'

const ImageStyled = styled.img`
  box-sizing: border-box;
  flex: 1 1;
  overflow: hidden;
  display: block;

  ${styles.compose(
    styles.objectFit,
    styles.space,
    styles.layout
  )};
`

const Image = (props) => (
  <ImageStyled
    {...props}
  />
)

Image.propTypes = {
  ...types.objectFit,
  ...types.space,
  ...types.layout,
  src: types.string.isRequired,
  alt: types.string.isRequired
}

export default Image
