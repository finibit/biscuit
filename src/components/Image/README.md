### Usage

#### Space

```jsx
<Image src="https://picsum.photos/200" margin={2} padding={2} />
```

#### Size

```jsx
<Image src="https://picsum.photos/200" width="50px" height="50px" />
```
