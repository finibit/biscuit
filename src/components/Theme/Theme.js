import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
