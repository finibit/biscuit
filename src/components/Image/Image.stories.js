import React from 'react'
import { storiesOf } from '@storybook/react'
import Wrapper from '../../Wrapper'
import Image from './Image'

storiesOf('Image', module)
  .addDecorator(Wrapper)
  .add('with size', () => (
    <Image src="https://picsum.photos/128" alt="picsum" size={64} />
  ))
  .add('with space', () => (
    <>
      <Image src="https://picsum.photos/128" alt="picsum" p={4} />
      <Image src="https://picsum.photos/128" alt="picsum" m={4} />
      <Image src="https://picsum.photos/128" alt="picsum" p={4} />
    </>
  ))