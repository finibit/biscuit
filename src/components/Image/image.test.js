import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Image from './Image'

describe('Image', () => {
	test('with margin', () => {
		const component = renderer.create((
			<Theme>
				<Image src="" alt="" m={1} />
				<Image src="" alt="" m={32} />
				<Image src="" alt="" m="5%" />
				<Image src="" alt="" mt={1} />
				<Image src="" alt="" mt={32} />
				<Image src="" alt="" mt="5%" />
				<Image src="" alt="" ml={1} />
				<Image src="" alt="" ml={32} />
				<Image src="" alt="" ml="5%" />
				<Image src="" alt="" mb={1} />
				<Image src="" alt="" mb={32} />
				<Image src="" alt="" mb="5%" />
				<Image src="" alt="" mr={1} />
				<Image src="" alt="" mr={32} />
				<Image src="" alt="" mr="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('with padding', () => {
		const component = renderer.create((
			<Theme>
				<Image src="" alt="" p={1} />
				<Image src="" alt="" p={32} />
				<Image src="" alt="" p="5%" />
				<Image src="" alt="" pt={1} />
				<Image src="" alt="" pt={32} />
				<Image src="" alt="" pt="5%" />
				<Image src="" alt="" pl={1} />
				<Image src="" alt="" pl={32} />
				<Image src="" alt="" pl="5%" />
				<Image src="" alt="" pb={1} />
				<Image src="" alt="" pb={32} />
				<Image src="" alt="" pb="5%" />
				<Image src="" alt="" pr={1} />
				<Image src="" alt="" pr={32} />
				<Image src="" alt="" pr="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
  })
  
	test('with size', () => {
		const component = renderer.create((
			<Theme>
				<Image src="" alt="" size="sm" />
				<Image src="" alt="" size={64} />
				<Image src="" alt="" size="5%" />
				<Image src="" alt="" width="sm" />
				<Image src="" alt="" width={64} />
				<Image src="" alt="" width="5%" />
				<Image src="" alt="" height="sm" />
				<Image src="" alt="" height={64} />
				<Image src="" alt="" height="5%" />
				<Image src="" alt="" minWidth={1} />
				<Image src="" alt="" minWidth={64} />
				<Image src="" alt="" minWidth="5%" />
				<Image src="" alt="" minHeight={1} />
				<Image src="" alt="" minHeight={64} />
				<Image src="" alt="" minHeight="5%" />
				<Image src="" alt="" maxWidth={1} />
				<Image src="" alt="" maxWidth={32} />
				<Image src="" alt="" maxWidth="5%" />
				<Image src="" alt="" maxHeight={1} />
				<Image src="" alt="" maxHeight={32} />
				<Image src="" alt="" maxHeight="5%" />
			</Theme>
		))
		expect(component.toJSON()).toMatchSnapshot()
  })
})