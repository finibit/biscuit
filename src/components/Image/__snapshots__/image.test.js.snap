// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image with margin 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin: 8px;
}

.c1 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin: 32px;
}

.c2 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin: 5%;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-top: 8px;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-top: 32px;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-top: 5%;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-left: 8px;
}

.c7 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-left: 32px;
}

.c8 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-left: 5%;
}

.c9 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-bottom: 8px;
}

.c10 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-bottom: 32px;
}

.c11 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-bottom: 5%;
}

.c12 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-right: 8px;
}

.c13 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-right: 32px;
}

.c14 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  margin-right: 5%;
}

<div>
  <img
    alt=""
    className="c0"
    src=""
  />
  <img
    alt=""
    className="c1"
    src=""
  />
  <img
    alt=""
    className="c2"
    src=""
  />
  <img
    alt=""
    className="c3"
    src=""
  />
  <img
    alt=""
    className="c4"
    src=""
  />
  <img
    alt=""
    className="c5"
    src=""
  />
  <img
    alt=""
    className="c6"
    src=""
  />
  <img
    alt=""
    className="c7"
    src=""
  />
  <img
    alt=""
    className="c8"
    src=""
  />
  <img
    alt=""
    className="c9"
    src=""
  />
  <img
    alt=""
    className="c10"
    src=""
  />
  <img
    alt=""
    className="c11"
    src=""
  />
  <img
    alt=""
    className="c12"
    src=""
  />
  <img
    alt=""
    className="c13"
    src=""
  />
  <img
    alt=""
    className="c14"
    src=""
  />
</div>
`;

exports[`Image with padding 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding: 8px;
}

.c1 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding: 32px;
}

.c2 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding: 5%;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-top: 8px;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-top: 32px;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-top: 5%;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-left: 8px;
}

.c7 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-left: 32px;
}

.c8 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-left: 5%;
}

.c9 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-bottom: 8px;
}

.c10 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-bottom: 32px;
}

.c11 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-bottom: 5%;
}

.c12 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-right: 8px;
}

.c13 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-right: 32px;
}

.c14 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  padding-right: 5%;
}

<div>
  <img
    alt=""
    className="c0"
    src=""
  />
  <img
    alt=""
    className="c1"
    src=""
  />
  <img
    alt=""
    className="c2"
    src=""
  />
  <img
    alt=""
    className="c3"
    src=""
  />
  <img
    alt=""
    className="c4"
    src=""
  />
  <img
    alt=""
    className="c5"
    src=""
  />
  <img
    alt=""
    className="c6"
    src=""
  />
  <img
    alt=""
    className="c7"
    src=""
  />
  <img
    alt=""
    className="c8"
    src=""
  />
  <img
    alt=""
    className="c9"
    src=""
  />
  <img
    alt=""
    className="c10"
    src=""
  />
  <img
    alt=""
    className="c11"
    src=""
  />
  <img
    alt=""
    className="c12"
    src=""
  />
  <img
    alt=""
    className="c13"
    src=""
  />
  <img
    alt=""
    className="c14"
    src=""
  />
</div>
`;

exports[`Image with size 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 32px;
  height: 32px;
}

.c1 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 64px;
  height: 64px;
}

.c2 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 5%;
  height: 5%;
}

.c3 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 32px;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 64px;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  width: 5%;
}

.c6 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  height: 32px;
}

.c7 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  height: 64px;
}

.c8 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  height: 5%;
}

.c9 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-width: 1px;
}

.c10 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-width: 64px;
}

.c11 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-width: 5%;
}

.c12 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-height: 1px;
}

.c13 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-height: 64px;
}

.c14 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  min-height: 5%;
}

.c15 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-width: 1px;
}

.c16 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-width: 32px;
}

.c17 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-width: 5%;
}

.c18 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-height: 1px;
}

.c19 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-height: 32px;
}

.c20 {
  box-sizing: border-box;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  display: block;
  max-height: 5%;
}

<div>
  <img
    alt=""
    className="c0"
    size="sm"
    src=""
  />
  <img
    alt=""
    className="c1"
    size={64}
    src=""
  />
  <img
    alt=""
    className="c2"
    size="5%"
    src=""
  />
  <img
    alt=""
    className="c3"
    src=""
    width="sm"
  />
  <img
    alt=""
    className="c4"
    src=""
    width={64}
  />
  <img
    alt=""
    className="c5"
    src=""
    width="5%"
  />
  <img
    alt=""
    className="c6"
    height="sm"
    src=""
  />
  <img
    alt=""
    className="c7"
    height={64}
    src=""
  />
  <img
    alt=""
    className="c8"
    height="5%"
    src=""
  />
  <img
    alt=""
    className="c9"
    src=""
  />
  <img
    alt=""
    className="c10"
    src=""
  />
  <img
    alt=""
    className="c11"
    src=""
  />
  <img
    alt=""
    className="c12"
    src=""
  />
  <img
    alt=""
    className="c13"
    src=""
  />
  <img
    alt=""
    className="c14"
    src=""
  />
  <img
    alt=""
    className="c15"
    src=""
  />
  <img
    alt=""
    className="c16"
    src=""
  />
  <img
    alt=""
    className="c17"
    src=""
  />
  <img
    alt=""
    className="c18"
    src=""
  />
  <img
    alt=""
    className="c19"
    src=""
  />
  <img
    alt=""
    className="c20"
    src=""
  />
</div>
`;
